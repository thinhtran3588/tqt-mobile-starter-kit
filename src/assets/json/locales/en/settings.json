{
  "author": "Author",
  "betaUser": "Beta user",
  "copyright": "Copyright",
  "darkTheme": "Dark theme",
  "generalInfo": "General info",
  "joinBetaUserConfirmation": "Do you want to be a beta user & get update more frequently?",
  "language": "Language",
  "leaveBetaUserConfirmation": "Do you want to leave beta program?",
  "primaryColor": "Primary color",
  "privacyPolicy": "Privacy Policy",
  "signIn": "Sign in",
  "signOut": "Sign out",
  "tos": "Terms Of Service",
  "useSystemTheme": "Use system theme",
  "version": "Version"
}
