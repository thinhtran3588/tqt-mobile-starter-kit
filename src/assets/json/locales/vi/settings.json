{
  "author": "Tác giả",
  "copyright": "Bản quyền",
  "darkTheme": "Theme tối",
  "generalInfo": "Thông tin chung",
  "primaryColor": "Màu chính",
  "useSystemTheme": "Dùng theme hệ thống",
  "version": "Phiên bản"
}
