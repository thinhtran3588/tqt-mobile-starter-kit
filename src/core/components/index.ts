export * from '@tqt/mobile/lib/components';
export * from './icon/icon.component';
export {
  Provider as PaperProvider,
  DefaultTheme as PaperDefaultTheme,
  Colors,
  Button,
  Appbar,
  Card,
  ActivityIndicator,
  Avatar,
  Badge,
} from 'react-native-paper';
export {StatusBar, Alert, ScrollView} from 'react-native';
