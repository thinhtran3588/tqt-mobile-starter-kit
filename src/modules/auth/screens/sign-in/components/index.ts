export * from './social-sign-in/social-sign-in.component';
