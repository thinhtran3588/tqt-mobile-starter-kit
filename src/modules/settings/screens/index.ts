export * from './settings/settings.screen';
export * from '../../../core/screens/web-view/web-view.screen';
