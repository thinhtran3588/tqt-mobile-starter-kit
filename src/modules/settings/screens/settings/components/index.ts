export * from './theme-setting/theme-setting.component';
export * from './primary-color-setting/primary-color-setting.component';
export * from './general-info/general-info.component';
export * from './language-setting/language-setting.component';
