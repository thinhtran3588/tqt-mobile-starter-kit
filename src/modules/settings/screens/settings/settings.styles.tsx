import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  header: {
    // backgroundColor: '#fff',
  },
  item: {
    padding: 0,
  },
  itemTitle: {
    fontWeight: 'bold',
  },
  switchContainer: {
    display: 'flex',
    justifyContent: 'center',
    paddingRight: 10,
  },
});
