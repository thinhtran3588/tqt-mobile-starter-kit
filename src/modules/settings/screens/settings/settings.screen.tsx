import React from 'react';
import {useTranslation} from 'react-i18next';
import {Text, List, Layout, ScrollView} from '@core/components';
import {ThemeSetting, GeneralInfo, LanguageSetting} from './components';

export const SettingsScreen = (): JSX.Element => {
  const {t} = useTranslation('settings');
  return (
    <Layout headerTitle={t('common:settings')}>
      <ScrollView>
        <ThemeSetting />
        <LanguageSetting />
        <List.Subheader>
          <Text type='h4'>{t('generalInfo')}</Text>
        </List.Subheader>
        <GeneralInfo />
      </ScrollView>
    </Layout>
  );
};
