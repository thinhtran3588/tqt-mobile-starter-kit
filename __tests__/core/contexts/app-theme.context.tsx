import React from 'react';
import {render, toJSON} from '@test-utils';
import {ThemeSetting} from '@settings/screens/settings/components';

it('renders with system light theme', async () => {
  const {baseElement} = render(<ThemeSetting />);
  expect(toJSON(baseElement)).toMatchSnapshot();
});

it('renders with system dark theme', async () => {
  jest.mock('react-native/Libraries/Utilities/useColorScheme', () => ({
    default: () => 'dark',
  }));
  const {baseElement} = render(<ThemeSetting />);
  expect(toJSON(baseElement)).toMatchSnapshot();
});
