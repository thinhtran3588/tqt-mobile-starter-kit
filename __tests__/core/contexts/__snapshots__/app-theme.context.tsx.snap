// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with system dark theme 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <TouchableHighlight
        descriptionNumberOfLines={2}
        disabled={true}
        style={
          Array [
            false,
            Array [
              Object {
                "padding": 8,
              },
              Array [
                Object {
                  "padding": 0,
                },
                undefined,
              ],
            ],
          ]
        }
        underlayColor="rgba(255, 255, 255, 0.16)"
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 8,
                  "width": 40,
                },
                undefined,
              ]
            }
          >
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰔎
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "marginVertical": 6,
                  "paddingLeft": 8,
                },
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "fontSize": 16,
                    },
                    Object {
                      "color": "rgba(255, 255, 255, 0.87)",
                    },
                    Array [
                      Object {
                        "fontWeight": "bold",
                      },
                      undefined,
                    ],
                  ],
                ]
              }
            >
              useSystemTheme
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "display": "flex",
                  "justifyContent": "center",
                  "paddingRight": 10,
                },
                Object {},
              ]
            }
          >
            <Switch
              testID="system-theme-switch"
              trackColor={
                Object {
                  "false": "",
                  "true": "#03dac6",
                }
              }
              value={true}
            />
          </View>
        </View>
      </TouchableHighlight>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(255, 255, 255, 0.12)",
              "height": 0.5,
            },
            undefined,
            undefined,
          ]
        }
      />
      <TouchableHighlight
        descriptionNumberOfLines={2}
        disabled={true}
        style={
          Array [
            false,
            Array [
              Object {
                "padding": 8,
              },
              Array [
                Object {
                  "padding": 0,
                },
                undefined,
              ],
            ],
          ]
        }
        underlayColor="rgba(255, 255, 255, 0.16)"
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 8,
                  "width": 40,
                },
                undefined,
              ]
            }
          >
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰔎
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "marginVertical": 6,
                  "paddingLeft": 8,
                },
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "fontSize": 16,
                    },
                    Object {
                      "color": "rgba(255, 255, 255, 0.87)",
                    },
                    Array [
                      Object {
                        "fontWeight": "bold",
                      },
                      undefined,
                    ],
                  ],
                ]
              }
            >
              darkTheme
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "display": "flex",
                  "justifyContent": "center",
                  "paddingRight": 10,
                },
                Object {},
              ]
            }
          >
            <Switch
              disabled={true}
              testID="dark-theme-switch"
              trackColor={
                Object {
                  "false": "",
                  "true": "#03dac6",
                }
              }
              value={false}
            />
          </View>
        </View>
      </TouchableHighlight>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(255, 255, 255, 0.12)",
              "height": 0.5,
            },
            undefined,
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`renders with system light theme 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <TouchableHighlight
        descriptionNumberOfLines={2}
        disabled={true}
        style={
          Array [
            false,
            Array [
              Object {
                "padding": 8,
              },
              Array [
                Object {
                  "padding": 0,
                },
                undefined,
              ],
            ],
          ]
        }
        underlayColor="rgba(0, 0, 0, 0.1)"
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 8,
                  "width": 40,
                },
                undefined,
              ]
            }
          >
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰔎
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "marginVertical": 6,
                  "paddingLeft": 8,
                },
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "fontSize": 16,
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 0.87)",
                    },
                    Array [
                      Object {
                        "fontWeight": "bold",
                      },
                      undefined,
                    ],
                  ],
                ]
              }
            >
              useSystemTheme
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "display": "flex",
                  "justifyContent": "center",
                  "paddingRight": 10,
                },
                Object {},
              ]
            }
          >
            <Switch
              testID="system-theme-switch"
              trackColor={
                Object {
                  "false": "",
                  "true": "#03dac4",
                }
              }
              value={true}
            />
          </View>
        </View>
      </TouchableHighlight>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.12)",
              "height": 0.5,
            },
            undefined,
            undefined,
          ]
        }
      />
      <TouchableHighlight
        descriptionNumberOfLines={2}
        disabled={true}
        style={
          Array [
            false,
            Array [
              Object {
                "padding": 8,
              },
              Array [
                Object {
                  "padding": 0,
                },
                undefined,
              ],
            ],
          ]
        }
        underlayColor="rgba(0, 0, 0, 0.1)"
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 8,
                  "width": 40,
                },
                undefined,
              ]
            }
          >
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰔎
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "marginVertical": 6,
                  "paddingLeft": 8,
                },
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "fontSize": 16,
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 0.87)",
                    },
                    Array [
                      Object {
                        "fontWeight": "bold",
                      },
                      undefined,
                    ],
                  ],
                ]
              }
            >
              darkTheme
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "display": "flex",
                  "justifyContent": "center",
                  "paddingRight": 10,
                },
                Object {},
              ]
            }
          >
            <Switch
              disabled={true}
              testID="dark-theme-switch"
              trackColor={
                Object {
                  "false": "",
                  "true": "#03dac4",
                }
              }
              value={false}
            />
          </View>
        </View>
      </TouchableHighlight>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.12)",
              "height": 0.5,
            },
            undefined,
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;
